lib/ffmpeg/vaapi/encoder.py

```diff
@@ -92,6 +92,7 @@ def check_output(self):
     ipbmode = 0 if vars(self).get("gop", 0) <= 1 else 1 if vars(self).get("bframes", 0) < 1 else 2
     ipbmsgs = [
       "Using intra frames only",
+      "Using intra and P-frames",
       "Using intra and P-frames|[L|l]ow delay|forward-prediction"
```

Comment:
Hmmm... the second one should already capture the one your trying to add.  Are you sure this fixes your problem?  Please provide full test case output.

Reply:
> Hmmm... the second one should already capture the one your trying to add. Are you sure this fixes your problem? Please provide full test case output.

 DEBUG: slash:   Output stream #0:0 (video): 240 frames encoded; 240 packets muxed (2044295 bytes);
[2023-01-04 02:50:59.560229] DEBUG: slash:   Total: 240 packets (2044295 bytes) muxed
[2023-01-04 02:50:59.578603] DEBUG: slash: [AVIOContext @ 0x56493619bcc0] Statistics: 331776000 bytes read, 0 seeks
[2023-01-04 02:50:59.580996] INFO: slash: DETAIL: time(ffmpeg:encode:1) = 3.3073s
[2023-01-04 02:50:59.588012] TRACE: slash.core.error: Error added: <Error: AssertionError: Possible incorrect IPB mode used
assert None is not None>
  /usr/local/lib/python3.8/dist-packages/slash/core/fixtures/parameters.py, line 36:
                    return func(*args, **kwargs)
  /opt/media/src/vaapi-fits/test/quality/ffmpeg/vaapi/avc.py, line 95:
        self.bdrate()
  /opt/media/src/vaapi-fits/test/quality/util.py, line 47:
            self.encode()
  /opt/media/src/vaapi-fits/lib/ffmpeg/encoderbase.py, line 225:
        self.check_output()
  /opt/media/src/vaapi-fits/lib/ffmpeg/vaapi/encoder.py, line 101:
        assert m is not None, "Possible incorrect IPB mode used"
[2023-01-04 02:50:59.618503] INFO: slash: DETAIL: time = 3.519641 seconds

Reply:
Please provide the full output for the test case.

Reply:
> Please provide the full output for the test case.

 ffmpeg -v verbose -hwaccel vaapi i /opt/assets/otc-media/yuv/RedKayak_1920x1080_30.yuv -vf 'format=nv12,hwupload' -an -c:v h264_vaapi -rc_mode VBR -profile:v main -compression_level 7 -g 2alidation.vaapi-fits-quality/results/quality/458fa2ac-8d7d-11ed-8e65-0242ac120002_0/_0.test.quality.ffmpeg.vaapi.avc/vbr_lp/458fa2ac-8d7d-11ed-8e65-0242ac1
ffmpeg version N-109457-geeb280f351 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
  configuration: --prefix=/opt/media/install --disable-static --enable-shared --enable-libdrm --enable-vaapi --enable-libmfx --disable-amf --disable-audiotable-nvdec --disable-nvenc --disable-omx --disable-omx-rpi --disable-rkmpp --disable-v4l2-m2m --disable-vdpau --disable-videotoolbox --enable-gpl --enable-
  libavutil      57. 43.100 / 57. 43.100
  libavcodec     59. 55.103 / 59. 55.103
  libavformat    59. 34.102 / 59. 34.102
  libavdevice    59.  8.101 / 59.  8.101
  libavfilter     8. 53.100 /  8. 53.100
  libswscale      6.  8.112 /  6.  8.112
  libswresample   4.  9.100 /  4.  9.100
  libpostproc    56.  7.100 / 56.  7.100
[AVHWDeviceContext @ 0x555876ab5680] libva: VA-API version 1.17.0
[AVHWDeviceContext @ 0x555876ab5680] libva: User environment variable requested driver 'iHD'
[AVHWDeviceContext @ 0x555876ab5680] libva: Trying to open /opt/media/install/lib/dri/iHD_drv_video.so
[AVHWDeviceContext @ 0x555876ab5680] libva: Found init function __vaDriverInit_1_17
[AVHWDeviceContext @ 0x555876ab5680] libva: va_openDriver() returns 0
[AVHWDeviceContext @ 0x555876ab5680] Initialised VAAPI connection: version 1.17
[AVHWDeviceContext @ 0x555876ab5680] VAAPI driver: Intel iHD driver for Intel(R) Gen Graphics - 22.6.5 ().
[AVHWDeviceContext @ 0x555876ab5680] Driver not found in known nonstandard list, using standard behaviour.
[rawvideo @ 0x555876ae1fc0] Estimating duration from bitrate, this may be inaccurate
Input #0, rawvideo, from '/opt/assets/otc-media/yuv/RedKayak_1920x1080_30.yuv':
  Duration: 00:00:19.00, start: 0.000000, bitrate: 746496 kb/s
  Stream #0:0: Video: rawvideo, 1 reference frame (I420 / 0x30323449), yuv420p, 1920x1080, 746496 kb/s, 30 tbr, 30 tbn
Stream mapping:
  Stream #0:0 -> #0:0 (rawvideo (native) -> h264 (h264_vaapi))
Press [q] to stop, [?] for help
[graph 0 input from stream 0:0 @ 0x555876bc6c00] w:1920 h:1080 pixfmt:yuv420p tb:1/30 fr:30/1 sar:0/1
[auto_scale_0 @ 0x555876bcab40] w:iw h:ih flags:'' interl:0
[Parsed_format_0 @ 0x555876bc1700] auto-inserting filter 'auto_scale_0' between the filter 'graph 0 input from stream 0:0' and the filter 'Parsed_format_0'
[auto_scale_0 @ 0x555876bcab40] w:1920 h:1080 fmt:yuv420p sar:0/1 -> w:1920 h:1080 fmt:nv12 sar:0/1 flags:0x0
[h264_vaapi @ 0x555876bbf2c0] Using input frames context (format vaapi) with h264_vaapi encoder.
[h264_vaapi @ 0x555876bbf2c0] Input surface format is nv12.
[h264_vaapi @ 0x555876bbf2c0] Using VAAPI profile VAProfileH264Main (6).
[h264_vaapi @ 0x555876bbf2c0] Using VAAPI entrypoint VAEntrypointEncSliceLP (8).
[h264_vaapi @ 0x555876bbf2c0] Using VAAPI render target format YUV420 (0x1).
[h264_vaapi @ 0x555876bbf2c0] RC mode: VBR.
[h264_vaapi @ 0x555876bbf2c0] RC target: 50% of 6000000 bps over 1000 ms.
[h264_vaapi @ 0x555876bbf2c0] RC buffer: 6000000 bits, initial fullness 4500000 bits.
[h264_vaapi @ 0x555876bbf2c0] RC framerate: 30/1 (30.00 fps).
[h264_vaapi @ 0x555876bbf2c0] Driver does not report any additional prediction constraints.
[h264_vaapi @ 0x555876bbf2c0] Using intra and P-frames (supported references: 3 / 0).
[h264_vaapi @ 0x555876bbf2c0] All wanted packed headers available (wanted 0xd, found 0x1f).
[h264_vaapi @ 0x555876bbf2c0] Using level 4.
Output #0, h264, to '/opt/media/src/applications.media.middleware.validation.vaapi-fits-quality/results/quality/458fa2ac-8d7d-11ed-8e65-0242ac120002_0/_0.t
  Metadata:
    encoder         : Lavf59.34.102
  Stream #0:0: Video: h264 (Main), 1 reference frame, vaapi(tv, progressive), 1920x1080 (0x0), q=2-31, 3000 kb/s, 30 fps, 30 tbn
    Metadata:
      encoder         : Lavc59.55.103 h264_vaapi
EOF in input file 0 q=-0.0 size=    5888kB time=00:00:16.66 bitrate=2894.1kbits/s speed=16.6x
Terminating demuxer thread 0
No more output streams to write to, finishing.
All streams finished for output file #0
Terminating muxer thread 0
[AVIOContext @ 0x555876bc28c0] Statistics: 7047761 bytes written, 0 seeks, 27 writeouts
frame=  570 fps=500 q=-0.0 Lsize=    6883kB time=00:00:18.96 bitrate=2972.7kbits/s speed=16.6x
video:6883kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000000%
Input file #0 (/opt/assets/otc-media/yuv/RedKayak_1920x1080_30.yuv):
  Input stream #0:0 (video): 570 packets read (1772928000 bytes); 570 frames decoded;
  Total: 570 packets (1772928000 bytes) demuxed
Output file #0 (/opt/media/src/applications.media.middleware.validation.vaapi-fits-quality/results/quality/458fa2ac-8d7d-11ed-8e65-0242ac120002_0/_0.test.q
  Output stream #0:0 (video): 570 frames encoded; 570 packets muxed (7047761 bytes);
  Total: 570 packets (7047761 bytes) muxed
[AVIOContext @ 0x555876ae25c0] Statistics: 1772928000 bytes read, 0 seeks

Reply:
How about just provide me a link to CI failure case?

Reply:
> How about just provide me a link to CI failure case?



> How about just provide me a link to CI failure case?

I have made new changes and want to get some suggestions and help from you

Reply:
> How about just provide me a link to CI failure case?
These failure cases were encountered when I added vaapi bdrate references 
https://github.com/intel-innersource/applications.media.middleware.validation.vaapi-fits-quality/pull/61/commits/0e37c82800213401b9afe102d4be55a438e1ffe5